cmd/rhc/connect_cmd.go

```diff
@@ -150,17 +164,146 @@ func beforeConnectAction(ctx *cli.Context) error {
 	return nil
 }
 
+func connectRHSM(ctx *cli.Context, connectResult *ConnectResult) {
```

Comment:
I'm not really happy these new methods are mutating their argument. It would be cleaner to have some kind of action struct that has the `connectResult` as its own variable. I.e.

```go
func (s *ConnectAction) TryRegisterRHSM(ctx *cli.Context) {
    returnedMsg, err := rhsm.RegisterRHSM(ctx, features.ConnectFeature.Enabled)
    if err != nil {
        s.result.RHSMConnected = false
        ...
```